File: src/main/java/io/github/toberocat/core/utility/events/faction/FactionLeaveEvent.java
package io.github.toberocat.core.utility.events.faction;

import io.github.toberocat.core.utility.factions.Faction;
import org.bukkit.entity.Player;

public class FactionLeaveEvent extends FactionEventCancelledable {

    protected Player player;

    public FactionLeaveEvent(Faction faction, Player player) {
        super(faction);
        this.player = player;
    }

    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }
}

File: improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/api/events/FactionLeaveEvent.kt
package io.github.toberocat.improvedfactions.api.events

import io.github.toberocat.improvedfactions.database.storage.FactionSnapshot
import io.github.toberocat.improvedfactions.database.storage.UserSnapshot
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

/**
 * Fired when a player leaves a faction.
 */
class FactionLeaveEvent(val faction: FactionSnapshot, val user: UserSnapshot) : Event(), Cancellable {
    private var isCancelled = false

    override fun getHandlers(): HandlerList = handlerList
    
    override fun isCancelled(): Boolean = isCancelled

    override fun setCancelled(cancel: Boolean) {
        isCancelled = cancel
    }

    companion object {
        @JvmStatic
        val handlerList = HandlerList()
    }
}
